File: 1475-final-prices-with-a-special-discount-in-a-shop/1475-final-prices-with-a-special-discount-in-a-shop.java
class Solution {
    public int[] finalPrices(int[] prices) {
        int n = prices.length;
        if(n<=1)
            return prices;
        int[] ans = new int[n];
        for(int i=0; i<n; i++){
            ans[i] = prices[i];
            for(int j=i+1; j<n; j++){
                if(prices[j]<=prices[i]){
                    ans[i] = prices[i] - prices[j];
                    break;
                }
            }
        }
        return ans;
    }
}

File: 1475-final-prices-with-a-special-discount-in-a-shop/1475-final-prices-with-a-special-discount-in-a-shop.cpp
class Solution {
public:
    vector<int> finalPrices(vector<int>& prices) {
        vector<int> result(prices.begin(),prices.end());
        stack<int>st;
        for(int i=0; i<prices.size();i++){
            while(!st.empty() and prices[i]<=prices[st.top()]){
              result[st.top()]-=prices[i];
              st.pop();
            }
            st.push(i);
        }
        return result;
    }
};